PythonTool/TagExpansion.py

```diff
@@ -114,20 +114,51 @@ def get_tag_lists(mb, element_type, element_id):
     return element_list, scalar_tags, vector_tags
 
 
-def create_database(mesh_file, mb_ref, mb_exp, elements, scal_tags, vec_tag, dir_name):
+def create_directory(mesh_file, main_dir_name = None):
     """
-    Expand the vector tag on each element in the given mesh data file. Write a
-    file to disk for each index with the corresponding scalar tag value.
+    Create a directory for the vector tag expansion.
 
     Input:
     ______
        mesh_file: str
            User supplied mesh file location.
-       mb_ref: Core
-           A PyMOAB core instance with a loaded data file for reference.
-       mb_exp: Core
-           A PyMOAB core instance with a loaded data file for expanding
-           vector tags.
+       main_dir_name: str
+           Optional user supplied name for main directory.
+
+    Returns:
+    ________
+       dir_name: str
+           The name of the directory in which to create and populate each
+           vector tag directory.
+    """
+
+    # Create a directory for the vector tag expansion files.
+    if main_dir_name is None:
+        input_list = mesh_file.split("/")
```

Comment:
It appears that `mesh_file` is only passed here for the sake of naming the new files created directory, but there is also an optional `main_dir_name` argument. Could this be rolled into 1 single argument called `output_dir`? The directory created can have the same root name as the files generated, so there is no real reason they have to be two separate arguments.

If anything you could add an optional boolean argument for overwriting files already present. For example, `True` to overwrite a directory that already exists or `False` to make a new directory (appended w/ a number or so).